src/transformers/generation/utils.py

```diff
@@ -444,10 +449,12 @@ def prepare_inputs_for_generation(
                     attention_mask,
                     sequence_length=sequence_length,
                     target_length=past_key_values.get_max_cache_shape(),
-                    dtype=self.get_output_embeddings().weight.dtype,
+                    dtype=self.dtype,
```

Comment:
for my own understanding, is this needed for multi-gpu setting?

Reply:
This change is needed because `moshi` doesn't have `get_output_embeddings()`, and creating `get_output_embeddings()` there would be a bit ambiguous.

`self.dtype` works just as fine and is more versatile :)

Reply:
oh dtype, not device 😄